DOCSP-44696-intellij-side-panel (#17)
mongodb/docs-mongodb-intellij: source/connect.txt

.. procedure:: 
   :style: normal 

   .. step:: Open the MongoDB side panel.

      To open the MongoDB side panel, click the leaf icon on the far-right menu 

>       bar of IntelliJ.

Reviewer: (minor suggestion)
For a very slight minimalism improvement "... click the leaf icon on IntelliJ's far-right menu bar." Or "the IntelliJ far-right menu bar"

>       .. image:: /images/connect/intellij-mongodb-side.png 

Reviewer: (image) 
I _think_ the red dot in the middle of the leaf is something we'd remove, no?

Author: The red dot was there in the UI before, and not added by us!

         :alt: Click the Database Explorer icon.
         :width: 600px

   .. step:: Add a MongoDB connection.

      To open the :guilabel:`Data Sources and Drivers` modal, 
      click :guilabel:`Add a MongoDB Data Source`.

      Paste your :manual:`connection string </reference/connection-string/>` 
      directly in the :guilabel:`URL` field near the bottom of the 
      :guilabel:`Data Sources and Drivers` tab.
      
      Ensure that your connection string specifies the port that your
      instance is running on. 
      
      Click :guilabel:`OK`.

>    .. step:: View insights for your MongoDB deployment.

Reviewer: (image)
For the image in this step, there is a significant chunk of empty space at the bottom of the screenshot. Is it possible to crop so less of that is displayed?

>    .. step:: View insights for your MongoDB deployment.

Reviewer: (question) 
It seems like this might be optional as well?

Author: I opted to not make this optional because the insights are a significant feature for the IntelliJ plugin.

      After you connect to your deployment, you can see all the generated 

>       insights for your workspace. 

Reviewer: (details)
Can you include a short description of what type of information is delivered in the insights?

      The {+intellij-short+} includes four options to view your 
      :guilabel:`Insights`:
      
      - :guilabel:`All insights`: Includes insights for all packages in your 
        workspace.

      - :guilabel:`Recommended Insights`: Includes insights for the most common 
        parent package. This ensures that insights are relevant to all files 
        and their shared context within the broader structure of your 
        application.

      - :guilabel:`Current File`: Includes insights to just the currently open 
        file.

      - :guilabel:`Current Query`: Includes insights for the query where the 
        keyboard cursor is located.

   .. step:: (*Optional*) Connect to another MongoDB deployment. 

      To add another data source, click the :guilabel:`+` icon, 
      click :guilabel:`Data Source`, and select :guilabel:`MongoDB` from the 
      drop-down menu.

>       .. image:: /images/connect/intellij-data-source.png

Reviewer: (image suggestion)
I think you can omit the arrow here. First, because users should be able to see it with just the highlighted box. Second, because it looks a little messy when covering other GUI text. WDYT?

Author: Will do! Updating.

         :width: 600px

   .. step:: Add a MongoDB connection.

      To open the :guilabel:`Data Sources and Drivers` modal, 
      click :guilabel:`Add a MongoDB Data Source`.

      Paste your :manual:`connection string </reference/connection-string/>` 
      directly in the :guilabel:`URL` field near the bottom of the 
      :guilabel:`Data Sources and Drivers` tab.
      
      Ensure that your connection string specifies the port that your
      instance is running on. 
      
      Click :guilabel:`OK`.

   .. step:: View insights for your MongoDB deployment.

      After you connect to your deployment, you can see all the generated 
      insights for your workspace, which include index performance warnings and 
      validation errors. 

      The {+intellij-short+} includes four options to view your 
      :guilabel:`Insights`:
      
      - :guilabel:`All insights`: Includes insights for all packages in your 
        workspace.

      - :guilabel:`Recommended Insights`: Includes insights for the most common 
        parent package. This ensures that insights are relevant to all files 
        and their shared context within the broader structure of your 

>         application.

Reviewer 2: I think the first statement should specify - *... for the most common parent package of files opened in the IDE* (or something along similar lines). It is necessary to specify that because if there are no files then there is no common package and hence no insights.

Author: Gotcha! Updating with the wording you suggested.

>       - :guilabel:`Current File`: Includes insights to just the currently open 
>         file.

Reviewer 2: ```suggestion
      - :guilabel:`Current File`: Includes insights from the currently open 
        file.
```

nit: I am not that good with my English so please take this with a grain of salt. It just felt odd reading that hence the suggestion.

Author: No worries! If anything, we want to simplify the English as much as possible for non-native English speakers :-) Updating!
